misenhower/homebridge-wyze-connected-home
# Fix: Setting groups of lights by a scene

### Problem
When setting groups of lights via a scene, it doesn't always set all the lights to the correct power state, brightness, _and_ color. This is due to a couple of possible reasons:
- Requests are sent simultaneously to Wyze, but Wyze requires a delay between each request to recognize the command.
- Requests are sent to Wyze out of order. Sometimes it sets the brightness, and then the power is turned on.

### How to test/recreate
- Set up a scene with 2 or more wyze bulbs setting brightness to 100% and color temp to white for both
- Set up a scene with 2 or more wyze bulbs setting brightness to 20% and color temp to yellow for both

Then try toggling between the two scenes. You'll see it wont always work, but with delays, it does.

### Solution
This adds a small async delay to the brightness & color, which guarantees setting the power state first. This seems to have fixed the underlying issue, but I'm not sure it is the best approach. I'm new to writing homebridge plugins :)

~~Also, this adds the `package-lock.json` to the repo. Not sure why that was not included?~~

This fixes https://github.com/misenhower/homebridge-wyze-connected-home/issues/19
